rotkehlchen/tests/exchanges/test_binance.py

```diff
@@ -628,3 +651,139 @@ def check_permutations_of_input_invalid_data(deposits, withdrawals):
         deposits=input_deposits,
         withdrawals=empty_withdrawals,
     )
+
+
+def test_binance_query_deposits_withdrawals_gte_90_days(function_scope_binance):
+    """Test the not so happy case of binance deposit withdrawal query
+
+    From `start_ts` to `end_ts` there is a difference gte 90 days, which forces
+    to request using a time delta (from API_TIME_INTERVAL_CONSTRAIN_TS).
+
+    This test implementation must mock 4 requests instead of 2.
+    """
+    start_ts = 0  # Defaults to BINANCE_LAUNCH_TS
+    end_ts = BINANCE_LAUNCH_TS + API_TIME_INTERVAL_CONSTRAIN_TS  # eq 90 days after
+    binance = function_scope_binance
+
+    def get_time_delta_deposit_result():
+        results = [
+            BINANCE_DEPOSITS_HISTORY_RESPONSE,
+            '{}',
+        ]
+        for result in results:
+            yield result
+
+    def get_time_delta_withdraw_result():
+        results = [
+            '{}',
+            BINANCE_WITHDRAWALS_HISTORY_RESPONSE,
+        ]
+        for result in results:
+            yield result
+
+    def mock_get_deposit_withdrawal(url):  # pylint: disable=unused-argument
+        if 'deposit' in url:
+            response_str = next(get_deposit_result)
+        else:
+            response_str = next(get_withdraw_result)
+
+        return MockResponse(200, response_str)
+
+    get_deposit_result = get_time_delta_deposit_result()
+    get_withdraw_result = get_time_delta_withdraw_result()
+
+    with patch.object(binance.session, 'get', side_effect=mock_get_deposit_withdrawal):
+        movements = binance.query_online_deposits_withdrawals(
+            start_ts=Timestamp(start_ts),
+            end_ts=Timestamp(end_ts),
+        )
+
+    errors = binance.msg_aggregator.consume_errors()
+    warnings = binance.msg_aggregator.consume_warnings()
+    assert len(errors) == 0
+    assert len(warnings) == 0
+
+    assert len(movements) == 4
+
+
+@pytest.mark.freeze_time(datetime(2020, 11, 24, 3, 14, 15))
+def test_api_query_dict_calls_with_time_delta(function_scope_binance):
```

Comment:
I am a bit confused. What is this test testing that the other test you made is not? Can you explain a bit?

Reply:
mmm... now I see them both again, I think I could merge them in one.

Not sure if I could justify to keep them both because the previous test would pass with both your time range implementation and my time range implementation. Whilst this test will only pass with my implementation.

Reply:
Ah, I remember now. The thing is as I'm patching `api_query_dict`. I'd need a more complex implementation for asserting the 4 "movements". This test is quite specific for the calls arguments.